src/backend/distributed/executor/adaptive_executor.c

```diff
@@ -2751,7 +2752,17 @@ ConnectionStateMachine(WorkerSession *session)
 					break;
 				}
 
+				int beforePollSocket = PQsocket(connection->pgConn);
 				PostgresPollingStatusType pollMode = PQconnectPoll(connection->pgConn);
+
+				int afterPollSocket = PQsocket(connection->pgConn);
+
+				if (beforePollSocket != afterPollSocket)
```

Comment:
we trigger this with the failure tests `SELECT citus.mitmproxy('conn.onAuthenticationOk().kill()');`. Though, I want to reproduce the epoll issue mentioned on #3193

Reply:
A clarification: The above triggers this case, however as the connection is already going to fail,  we cannot trigger `epoll_ctl` error in the next iteration.